### PR TITLE
Start using SolverTools

### PR DESCRIPTION
**Pull request by abelsiqueira**

### CODE REVIEW

File: src/NLPModelsIpopt.jl [hunk 1]
```diff
@@ -2,7 +2,27 @@ module NLPModelsIpopt
...
+                            #-3 => Error in step computation
+                            -4 => :max_time,
+                            #-10 => Not enough degress of freedom
+                            #-11 => Invalid problem definition
+                            #-12 => Invalid option
+                            #-13 => Invalid number detected
+                            -100 => :exception,
+                            -101 => :exception,
+                            -102 => :exception,
+                            -199 => :exception)
```
Comment (dpo) [inline src/NLPModelsIpopt.jl]: Shouldn't we grab IPOPT.jl's `ApplicationReturnStatus` and override the keys that we support more generally?

Comment (abelsiqueira) [reply src/NLPModelsIpopt.jl]: Do you mean creating more keys in `STATUSES` in `SolverTools`?

Comment (dpo) [reply src/NLPModelsIpopt.jl]: I meant to use their `ApplicationReturnStatus` and rename the statuses that we support (0, 1, 2, 3 and others), but keep the status that they define (which is IPOPT's) for the others (especially the last few exceptions).

Comment (abelsiqueira) [reply src/NLPModelsIpopt.jl]: I see. However, `GenericExecutionStats` only allow status that are in `STATUSES`, so we have to choose:
- we add those to `STATUSES`;
- change `GenericExecutionStats`;
- do what I did, the translation.

Adding a few is also an option and translating others is also an option.

Notice that the message is not lost, I keep it in `solver_specific[:internal_msg]` [here](https://github.com/JuliaSmoothOptimizers/NLPModelsIpopt.jl/pull/9/files#diff-9206df8d392bada610e908fbb4c6c48bR120).

Comment (dpo) [reply src/NLPModelsIpopt.jl]: Right. Ok, that's fine.

**Comment by dpo** [comment]
Thank you.

**Pull request merged**